add task model

david135 opened the pull request:
adding Task model according to #139.

david135 committed:
add task model

kgf commented on app/models/task.rb:
```diff
@@ -0,0 +1,60 @@
+  # This is used to give a sortable numeric value based on the order of the above array.
+  STATUS_CHOICES = STATUS_CHOICES_ARRAY.map.with_index { |v, i| [v, i] }.to_h
+
+  def status
```
To be clear to others who may be watching, this is only for testing. As we build out the other models needed for tracking events, this will go away.

kgf commented:
I would like to adjust the UI on this.
![screen shot 2016-09-29 at 11 57 17 am](https://cloud.githubusercontent.com/assets/2431417/18961729/fb7970de-863b-11e6-809a-88d868d426ca.png)
I think we could use more horizontal space for the data, and lose more of the labels.
So maybe 
**Category** Title at Location
(start time to end time duration)
Status (Available count in green/unavailable count in red/unknown count in yellow)
Then the tables across all columns below that.

Thoughts ?

david135 commented:
I was aiming to make use of currently unused space.  Compare:

The previous display of an event (`development`):
![event-before task addition](https://cloud.githubusercontent.com/assets/8586431/18968067/8e0267f8-8655-11e6-8a26-e4cffef6f889.png)
with the display now (`task_model`):
![event-after task addition](https://cloud.githubusercontent.com/assets/8586431/18968075/9690a74a-8655-11e6-9b3f-9600fd0fba20.png)

It sounds like with your changes, we'd be down to just a few lines at the top.

Here's my draft of that (`task_model_update1` - not yet submitted):
![event-task addition-update 1](https://cloud.githubusercontent.com/assets/8586431/18967674/252cfe9c-8654-11e6-8b26-8892a3e1f2f3.png)

I added the word personnel, to indicate the basic meaning of the numbers.  I styled the numbers as labels, because when simply coloring the numbers, one could barely detect the colors.

I put the time display into a helper function.  I envision someone improving it so that the date isn't repeated unless necessary, etc., but will leave that for another time.

I also moved the comments/created/last modified info back to the bottom (where they had been in `development`), where one will only find them if looking for them.

Is this on the way to what you have in mind?

(In general, I wonder about scarcity of horizontal space vs vertical space.  It is sometimes a burden on the user to have to scroll down, and to the extent that we can fit what is necessary on a single "page" without scrolling, that may be a worthy aim.)

kgf commented:
That looks MUCH better, and is very close to what I had imagined. We should be sure to color the rows in the tables the same way. ( Make everyone in the Not Responded table yellow ) This would reinforce the meaning of the colors. A tooltip on those numbers would be helpful too.
Thanks !

kgf commented:
How hard would it be to hide the pagination when not needed ?

david135 commented:
It isn't as easy as a configuration option.  It sounds like we'd need a JS callback to check the number.  Are we ever going to have more than a hundred entries for any of these tables?  We could disable paging by default and only enable it for tables we think will be large.

kgf commented:
I would say disable the paging then. It will be unusual for us to have that many rows in a table.

david135 committed:
requested (and other) improvements

kgf commented on app/helpers/application_helper.rb:
```diff
@@ -113,6 +119,14 @@ def format_datetime_value(datetime)
   end
 
+  def format_time_span(start_time, end_time, duration = nil)
+    if duration
```
If you have the start time and the end time, why not calculate the duration ?

kgf commented on spec/support/authorization_view_helper.rb:
```diff
@@ -33,4 +33,38 @@ def get_basic_editor_views(model_name, things_to_look_for)
+      @sample_object = create(nested_model)
+      index_model_path = url_for(action: :index, controller: nested_model_controller)
+      visit index_model_path
+      expect(page).to have_content("LIMS") # This is in the nav bar
```
Not going to stop the PR for this, but we should be starting to filter these out. The static text "LIMS" is going away.
